cmd/main.go

```diff
@@ -32,13 +32,12 @@ var globalFlags = []cli.Flag{
 	cli.StringFlag{
 		Name:  "config-dir, C",
 		Value: getConfigDir(),
-		Usage: func() string {
-			usage := "Path to configuration directory."
-			if getConfigDir() == "" {
-				usage = usage + "  This option must be set."
-			}
-			return usage
-		}(),
+		Usage: "[DEPRECATED] Path to legacy configuration directory.",
+	},
+	cli.StringFlag{
+		Name:  "certs-dir, S",
+		Value: getCertsDir(),
+		Usage: "Path to certs directory.",
```

Comment:
I have below thoughts on config directories.
1. config directories are constants i.e. only Gets.
2. default config directories need to be probed at `init()` and created at `init()`
3. user defined config directories are validated and created if they are passed.

If above points make sense, below code is good enough
```go

var globalConfigDir *configDir
var globalCertsDir *condfigDir
var globalCADir *configDir

var defaultMinioConfigDir *configDir
var defaultCertsDir *condfigDir
var defaultCADir *configDir

func init() {
    // find and set all default config dirs.
}

type configDir struct {
    path string
}

(dir *configDir) Dir() string {
    return dir.path
}

func newConfigDir(dir string) (&configDir, error) {
    // validate passed dir
    return &configDir{dir}, nil
}

// In `globalFlags`
 	cli.StringFlag{
 		Name:  "config-dir, C",
 		Value: defaultConfigDir.Dir(),
 		Usage: "[DEPRECATED] Path to legacy configuration directory.",
 	},
 	cli.StringFlag{
 		Name:  "certs-dir, S",
 		Value: defaultCertsDir.Dir(),
 		Usage: "Path to certs directory.",
 	},

// In command line argument validations.
globalConfigDir, err = newConfigDir(value-from-cmd)
globalCertsDir, err = newConfigDir(value-from-cmd)
globalCADir, err = newConfigDir(value-from-cmd)
```

Reply:
Actually the intention is we should not create unnecessary directories unless user specified them, so doing it in init() is not an option. In container environments we don't have write access everywhere. 

Current code is pretty much doing what you just wrote, let me see if I can simplify it.

Reply:
My previous comment is not about creating config directories.  Its about how config directory constants are created and used.  Its about implementing with simplicity.

Reply:
okay then let me take this approach and see where it leads me. @balamurugana will update the PR